graphics/src/MeshManager.cc

```diff
@@ -1648,3 +1660,82 @@ void MeshManager::SetAssimpEnvs()
     this->dataPtr->forceAssimp = true;
   }
 }
+
+//////////////////////////////////////////////////
+std::vector<SubMesh>
+MeshManager::ConvexDecomposition(const SubMesh *_subMesh,
+                                 std::size_t _maxConvexHulls,
+                                 std::size_t _voxelResolution)
+{
+  std::vector<SubMesh> decomposed;
+
+  if (!_subMesh)
+    return decomposed;
+
+  auto vertexCount = _subMesh->VertexCount();
+  auto indexCount = _subMesh->IndexCount();
+  auto triangleCount = indexCount / 3u;
+
+  float *points = new float[vertexCount * 3u];
+  for (std::size_t i = 0; i < vertexCount; ++i)
+  {
+    std::size_t idx = i * 3u;
+    points[idx] = _subMesh->Vertex(i).X();
+    points[idx + 1] = _subMesh->Vertex(i).Y();
+    points[idx + 2] = _subMesh->Vertex(i).Z();
+  }
+
+  uint32_t *indices = new uint32_t[indexCount];
+  for (std::size_t i = 0; i < indexCount; ++i)
+  {
+    indices[i] = _subMesh->Index(i);
+  }
+
+  VHACD::IVHACD *iface = VHACD::CreateVHACD_ASYNC();
+  VHACD::IVHACD::Parameters parameters;
+  parameters.m_maxConvexHulls = _maxConvexHulls;
+  parameters.m_resolution = _voxelResolution;
+  parameters.m_asyncACD = true;
+  iface->Compute(points, vertexCount, indices, triangleCount, parameters);
+  while (!iface->IsReady())
+  {
+    std::this_thread::sleep_for(std::chrono::nanoseconds(10000));
+  }
+
+  if (!iface->GetNConvexHulls())
+    return decomposed;
+
+  for (std::size_t n = 0; n < iface->GetNConvexHulls(); ++n)
+  {
+    VHACD::IVHACD::ConvexHull ch;
+    iface->GetConvexHull(n, ch);
+
+    SubMesh convexMesh;
+    for (std::size_t i = 0u; i < ch.m_points.size(); ++i)
+    {
+      const VHACD::Vertex &p = ch.m_points[i];
+      gz::math::Vector3d vertex(p.mX, p.mY, p.mZ);
+      convexMesh.AddVertex(vertex);
+
+      // add dummy normal
+      auto norm = vertex;
+      norm.Normalize();
```

Comment:
If `vertex` is (0,0,0), would `Normalize()` work okay?

Reply:
should be ok as there is a check for 0 in gz-math. 

We're not currently using normals for collisions so these should be good enough. If users do need valid normals, it can be done by calling `submesh.RecalculateNormals()`

Reply:
If it's just a dummy normal, why not just use a constant `0, 0, 1` or something and avoid the Normalize call? Also, if anyone ever uses the normals, it would be obvious something is wrong if all the normals are the same than if they're seemingly random.

Reply:
> it would be obvious something is wrong if all the normals are the same

if all normals are the same, it could just mean it's a valid plane. To be safe, I added the `RecalculateNormals` call to now generate valid normals.